aggregator_core/src/datastore/models.rs

```diff
@@ -1244,15 +1258,148 @@ where
 }
 
 /// Represents the state of a batch aggregation.
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
+pub enum BatchAggregationState<const SEED_SIZE: usize, A: vdaf::Aggregator<SEED_SIZE, 16>> {
```

Comment:
It might be strange to duplicate these values, but the `Scrubbed` state has different data, so I don't want to make state orthogonal from state-specific data.

I could introduce a structure to store the `aggregate_share`/`report_count`/`checksum`, but I'm not sure the additional level of indirection would be a win, and this might make things more inflexible if we decide we want to add fields to one of the states later.